provider/clusterautoscaler/cluster_autoscaler_resource.go

```diff
@@ -0,0 +1,640 @@
+							Attributes: map[string]schema.Attribute{
+								"type": schema.StringAttribute{
+									Required: true,
+								},
+								"range": rangeAttribute("limit number of GPU type", true, false),
+							},
+						},
+						Validators: []validator.List{
+							listvalidator.SizeAtLeast(1),
+						},
+					},
+				},
+			},
+			"scale_down": schema.SingleNestedAttribute{
+				Description: "Configuration of scale down operation.",
+				Optional:    true,
+				Attributes: map[string]schema.Attribute{
+					"enabled": schema.BoolAttribute{
+						Description: "Should cluster-autoscaler scale down the cluster.",
+						Optional:    true,
+					},
+					"unneeded_time": schema.StringAttribute{
+						Description: "How long a node should be unneeded before it is eligible for scale down.",
+						Optional:    true,
+					},
+					"utilization_threshold": schema.StringAttribute{
+						Description: "Node utilization level, defined as sum of requested resources divided " +
+							"by capacity, below which a node can be considered for scale down.",
+						Optional: true,
+						Validators: []validator.String{
+							stringFloatRangeValidator("utilization threshold validation", 0.0, 1.0),
+						},
+					},
+					"delay_after_add": schema.StringAttribute{
+						Description: "How long after scale up that scale down evaluation resumes.",
+						Optional:    true,
+						Validators:  []validator.String{durationStringValidator("delay after add validation")},
+					},
+					"delay_after_delete": schema.StringAttribute{
+						Description: "How long after node deletion that scale down evaluation resumes.",
+						Optional:    true,
+						Validators:  []validator.String{durationStringValidator("delay after delete validation")},
+					},
+					"delay_after_failure": schema.StringAttribute{
+						Description: "How long after scale down failure that scale down evaluation resumes.",
+						Optional:    true,
+						Validators:  []validator.String{durationStringValidator("delay after failure validation")},
+					},
+				},
+			},
+		},
+	}
+	return
+}
+func (r *ClusterAutoscalerResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	collection, ok := req.ProviderData.(*sdk.Connection)
+	if !ok {
+		resp.Diagnostics.AddError("Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *sdk.Connaction, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.collection = collection.ClustersMgmt().V1().Clusters()
+}
+
+func (r *ClusterAutoscalerResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	plan := &ClusterAutoscalerState{}
+	diags := request.Plan.Get(ctx, plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	resource := r.collection.Cluster(plan.Cluster.ValueString())
```

Comment:
we should also validate that there isn't an existing autoscaler for that cluster

Reply:
Im not sure about this... We should check before assuming.